packages/eslint-plugin/src/rules/contextual-decorator.ts

```diff
@@ -0,0 +1,123 @@
+import { TSESLint, TSESTree } from '@typescript-eslint/experimental-utils';
+
+import {
+  getDecoratorName,
+  isAngularClassDecorator,
+  isAngularInnerClassDecorator,
+  isNotNullOrUndefined,
+  AngularClassDecoratorKeys,
+  ANGULAR_CLASS_DECORATOR_MAPPER,
+} from '../utils/utils';
+import { createESLintRule } from '../utils/create-eslint-rule';
+
+type Options = [];
+export type MessageIds = 'contextualDecorator';
+export const RULE_NAME = 'contextual-decorator';
+
+export default createESLintRule<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Ensures that classes use contextual decorators in its body.',
+      category: 'Best Practices',
+      recommended: 'error',
```

Comment:
This has been marked as recommended but not added to the recommended config, so it is currently failing a new check in CI.

If this was recommended in codelyzer I'm happy for it to be recommended in angular-eslint (and in general I don't want to be a dictator about recommended configs, I spend a lot of my time outside of an Angular context at the moment so am not always the best judge anyway)

Reply:
Not sure if it was recommended in codelyzer :smile: I've marked it as "false" for now.